src/tests/ftest/control/dmg_nvme_scan_test.yaml

```diff
@@ -8,14 +8,14 @@ server_config:
   name: daos_server
   port: 10001
   transport_config:
-    allow_insecure: True
+    allow_insecure: False
 agent_config:
   transport_config:
-    allow_insecure: True
+    allow_insecure: False
 dmg:
   dmg_sub_command: storage
   storage:
     storage_sub_command: scan
-  insecure: True
+  insecure: False
   transport_config:
-    allow_insecure: True
+    allow_insecure: False
```

Comment:
Since we know this test passes with `allow_insecure: False` (https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-6015/1/artifact/Functional%20on%20CentOS%207/control/dmg_nvme_scan_test.py/) and with `allow_insecure: True` (https://build.hpdd.intel.com/job/daos-stack/job/daos/job/master/4840/artifact/Functional%20on%20CentOS%207/control/dmg_nvme_scan_test.py/), we should enable launch.py to control the setting fully with its `-ins` command line option. This means removing the current restriction of always running this test in secure mode and deleting lines 10-14 and lines 19-21.